tm2/pkg/sdk/baseapp.go

```diff
@@ -652,25 +652,34 @@ func (app *BaseApp) runMsgs(ctx Context, msgs []Msg, mode RunTxMode) (result Res
 		// Each message result's Data must be length prefixed in order to separate
 		// each result.
 		data = append(data, msgResult.Data...)
-		events = append(events, msgResult.Events...)
-		defer func() {
-			events = append(events, ctx.EventLogger().Events()...)
-			result.Events = events
-		}()
-		// TODO append msgevent from ctx. XXX XXX
+
+		// each msgs' event
+		var msgEvent []abci.Event
+
+		// handle each msg's ctx event
+		msgCtxEvents := ctx.EventLogger().Events()
+		for _, msgCtxEvent := range msgCtxEvents {
```

Comment:
I'm not sure what kind of performance overhead this has, but it can't be minimal